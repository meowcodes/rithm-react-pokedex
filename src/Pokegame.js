import React, { Component } from 'react';
import Pokedex from './Pokedex';
import pickRandom from './helper';

class Pokegame extends Component {
    static defaultProps = {
        pokemon: [
            { id: 4, name: 'Charmander', type: 'fire', base_experience: 62 },
            { id: 7, name: 'Squirtle', type: 'water', base_experience: 63 },
            { id: 11, name: 'Metapod', type: 'bug', base_experience: 72 },
            { id: 12, name: 'Butterfree', type: 'flying', base_experience: 178 },
            { id: 25, name: 'Pikachu', type: 'electric', base_experience: 112 },
            { id: 39, name: 'Jigglypuff', type: 'normal', base_experience: 95 },
            { id: 94, name: 'Gengar', type: 'poison', base_experience: 225 },
            { id: 133, name: 'Eevee', type: 'normal', base_experience: 65 }
        ]
    }

    render() {
        const pokemonsP1 = pickRandom(this.props.pokemon, 4);
        let expP1 = 0;
        for(let p of pokemonsP1){
            expP1 += p.base_experience
        }

        const pokemonsP2 = pickRandom(this.props.pokemon, 4);
        let expP2 = 0;
        for(let p of pokemonsP2){
            expP2 += p.base_experience
        }

    return (
        <div>
            <Pokedex pokemon={ pokemonsP1 } totalExp={ expP1 }/>
            <Pokedex pokemon={ pokemonsP2 } totalExp={ expP2 } />
        </div>
    );
  }
}

export default Pokegame;