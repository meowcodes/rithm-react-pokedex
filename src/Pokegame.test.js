import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Pokegame from "./Pokegame";

// smoke test
it("renders without crashing", function() {
  shallow(<Pokegame />);
  mount(<Pokegame />)
});

// snapshot test
it("matches snapshot", function() {
  let wrapper = shallow(<Pokegame pokemon={[{ id: 133, name: 'Eevee', type: 'normal', base_experience: 65 }]} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});