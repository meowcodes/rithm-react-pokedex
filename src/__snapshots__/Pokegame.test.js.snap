// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <Pokedex
    pokemon={
      Array [
        Object {
          "base_experience": 65,
          "id": 133,
          "name": "Eevee",
          "type": "normal",
        },
        Object {
          "base_experience": 65,
          "id": 133,
          "name": "Eevee",
          "type": "normal",
        },
        Object {
          "base_experience": 65,
          "id": 133,
          "name": "Eevee",
          "type": "normal",
        },
        Object {
          "base_experience": 65,
          "id": 133,
          "name": "Eevee",
          "type": "normal",
        },
      ]
    }
    totalExp={260}
  />
  <Pokedex
    pokemon={
      Array [
        Object {
          "base_experience": 65,
          "id": 133,
          "name": "Eevee",
          "type": "normal",
        },
        Object {
          "base_experience": 65,
          "id": 133,
          "name": "Eevee",
          "type": "normal",
        },
        Object {
          "base_experience": 65,
          "id": 133,
          "name": "Eevee",
          "type": "normal",
        },
        Object {
          "base_experience": 65,
          "id": 133,
          "name": "Eevee",
          "type": "normal",
        },
      ]
    }
    totalExp={260}
  />
</div>
`;
